uploadservice/src/main/java/net/gotev/uploadservice/UploadService.java

```diff
@@ -135,6 +135,7 @@
     private PowerManager.WakeLock wakeLock;
     private int notificationIncrementalId = 0;
     private static final Map<String, UploadTask> uploadTasksMap = new ConcurrentHashMap<>();
+    private static final Map<String, UploadTask> completedTasksMap = new ConcurrentHashMap<>();
```

Comment:
Having 2 maps here sets off some alarms for me - theoretically, it's possible to have an `UploadTask` both in `uploadTasksMap` and `completedTasksMap`.